symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java

```diff
@@ -0,0 +1,394 @@
+package com.symphony.bdk.core.service.user;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.gen.api.UserApi;
+import com.symphony.bdk.gen.api.UsersApi;
+import com.symphony.bdk.gen.api.model.Avatar;
+import com.symphony.bdk.gen.api.model.AvatarUpdate;
+import com.symphony.bdk.gen.api.model.DelegateAction;
+import com.symphony.bdk.gen.api.model.Disclaimer;
+import com.symphony.bdk.gen.api.model.Feature;
+import com.symphony.bdk.gen.api.model.StringId;
+import com.symphony.bdk.gen.api.model.UserDetail;
+import com.symphony.bdk.gen.api.model.UserFilter;
+import com.symphony.bdk.gen.api.model.UserSearchQuery;
+import com.symphony.bdk.gen.api.model.UserSearchResults;
+import com.symphony.bdk.gen.api.model.UserStatus;
+import com.symphony.bdk.gen.api.model.UserV2;
+import com.symphony.bdk.gen.api.model.V2UserDetail;
+import com.symphony.bdk.gen.api.model.V2UserList;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/**
+ * A class for implementing the user service.
+ *
+ * This service is used for retrieving information about a particular user,
+ * search users by ids, emails or usernames, perform some action related to
+ * user like:
+ * <p><ul>
+ *   <li>Add or remove roles from an user</li>
+ *   <li>Get or update avatar of an user</li>
+ *   <li>Get, assign or unassign disclaimer to an user</li>
+ *   <li>Get, update feature entitlements of an user</li>
+ *   <li>Get, update status of an user</li>
+ * </ul></p>
+ *
+ */
+@Slf4j
+public class UserService {
+
+  private final UserApi userApi;
+  private final UsersApi usersApi;
+  private final AuthSession authSession;
+
+  public UserService(UserApi userApi, UsersApi usersApi, AuthSession authSession) {
+    this.userApi = userApi;
+    this.usersApi = usersApi;
+    this.authSession = authSession;
+  }
+
+  /**
+   * Retrieve user details of a particular user.
+   * @param uid User Id
+   * @return Details of the user.
+   */
+  public V2UserDetail getUserDetailByUid(@NonNull Long uid) {
+    try {
+      return userApi.v2AdminUserUidGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Retrieve a list of users in the company (pod) by a filter.
+   * If no filter is given, retrieving all users in the company (pod).
+   *
+   * @param filter using to filter users by
+   * @return List of retrieved users
+   */
+  public List<V2UserDetail> listUsersDetailByFilter(@Nullable UserFilter filter) {
```

Comment:
Better to split this method in 2 different ones `listUsersDetail()` and `listUsersDetail(@Nonnull UserFilter filter)`